File: 268. Missing Number.py
class Solution:
    def missingNumber(self, nums: list[int]) -> int:
        return int((len(nums) + 1) * len(nums) / 2 - sum(nums))


newMissingNumber = Solution()
print(newMissingNumber.missingNumber([3, 0, 1]))

File: 268_missing_number.c
#include <stdio.h>

int missingNumber(int *nums, int numsSize)
{
    int xorAll = 0;
    int xorNums = 0;

    for (int i = 0; i <= numsSize; i++)
        xorAll ^= i;

    for (int i = 0; i < numsSize; i++)
        xorNums ^= nums[i];

    return xorAll ^ xorNums;
}

int main()

{
    int nums[3] = {3, 0, 1};
    int numsSize = sizeof(nums) / sizeof(nums[0]);

    int res = missingNumber(nums, numsSize);
    printf("%d", res);

    return 0;
}
